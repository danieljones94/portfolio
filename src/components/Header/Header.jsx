import React, { Component } from "react";
import styles from "./Header.module.scss";

class Header extends Component {
  state = {};
  render() {
    return (
      <header className={styles.headerContainer}>
        <section className={styles.titleContainer}>
          <div className={styles.leftSideHeader}>
            <h2>DANIEL</h2>
            <h3>WEB</h3>
          </div>
          <div className={styles.rightSideHeader}>
            <h2>JONES.</h2>
            <h3>DEVELOPER</h3>
          </div>
        </section>
        {/* <h3>WEB DEVELOPER</h3> */}
      </header>
    );
  }
}

export default Header;
