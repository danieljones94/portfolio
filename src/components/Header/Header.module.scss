.headerContainer {
  height: 110vh;
  font-family: "Montserrat", sans-serif;

  .titleContainer {
    display: flex;
    flex-direction: column;
    height: 100%;

    .title {
      height: 100%;
      width: 100%;
      background: linear-gradient(90deg, #ffffff 50%, #4a738c 50%);

      text-align: center;
      text-shadow: 0px 3px 0px #b2a98f, 0px 14px 10px rgba(0, 0, 0, 0.15),
        0px 24px 2px rgba(0, 0, 0, 0.1), 0px 34px 30px rgba(0, 0, 0, 0.1);
    }

    h1 {
      font-size: 4em;
      margin-top: 18%;
    }
    h2 {
      letter-spacing: 2px;
    }
  }

  // .leftSideHeader {
  //   width: 50%;
  //   background-color: white;
  //   color: #49728c;
  //   display: flex;
  //   flex-direction: column;
  // }

  // .leftSideHeader h2 {
  //   margin-top: 30%;
  //   margin-left: 57%;
  //   font-size: 4em;
  //   text-shadow: 0px 3px 0px #b2a98f, 0px 14px 10px rgba(0, 0, 0, 0.15),
  //     0px 24px 2px rgba(0, 0, 0, 0.1), 0px 34px 30px rgba(0, 0, 0, 0.1);
  // }

  // .leftSideHeader h3 {
  //   margin-left: 85%;
  //   font-size: 2em;
  //   animation: aniload 3s;
  //   // transform: scale(-1);
  //   // filter: FlipV;

  //   ::before {
  //     content: "WEB";
  //     transform: rotateY(180deg);
  //     background-image: linear-gradient(
  //       180deg,
  //       rgba(255, 255, 255, 0) 10%,
  //       rgba(255, 255, 255, 0.5)
  //     );
  //     -webkit-background-clip: text;
  //     color: transparent;
  //   }

  //   @keyframes aniload {
  //     from {
  //       transform: translate(-1000px, 0px);
  //     }
  //     to {
  //       transform: translate(0px, 0px);
  //     }
  //   }
  // }

  // .rightSideHeader {
  //   width: 50%;
  //   background-color: #4a738c;
  //   color: white;
  // }

  // .rightSideHeader h2 {
  //   margin-top: 30%;
  //   margin-left: 2vw;
  //   font-size: 4em;
  //   text-shadow: 0px 3px 0px #b2a98f, 0px 14px 10px rgba(0, 0, 0, 0.15),
  //     0px 24px 2px rgba(0, 0, 0, 0.1), 0px 34px 30px rgba(0, 0, 0, 0.1);
  // }

  // .rightSideHeader h3 {
  //   font-size: 2em;
  //   margin-left: 1vw;
  //   animation: aniload 3s;

  //   ::after {
  //     content: "DEVELOPER";
  //     transform: rotateY(180deg);
  //     background-image: linear-gradient(
  //       180deg,
  //       rgba(255, 255, 255, 0) 10%,
  //       rgba(255, 255, 255, 0.5)
  //     );
  //     -webkit-background-clip: text;
  //     color: transparent;
  //   }

  //   @keyframes aniload {
  //     from {
  //       transform: translate(1000px, 0px);
  //     }
  //     to {
  //       transform: translate(0px, 0px);
  //     }
  //   }
  // }

  // .webDeveloper {
  //   color: black;
  // }

  .scroll {
    display: block;
    text-align: center;
    margin: 2vw;
    font-size: 2.5em;
    color: rgb(80, 101, 110);
    &:hover {
      color: rgb(80 * 0.7, 101 * 0.7, 110 * 0.7);
      cursor: pointer;
    }
  }
}
