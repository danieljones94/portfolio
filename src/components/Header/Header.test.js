import React from "react";
import Header from "./Header";
import { mount } from "enzyme";
import ReactTestUtils from "react-dom/test-utils";

describe("Header tests", () => {
  let component;

  beforeEach(() => {
    component = mount(<Header />);
  });

  test("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  test("should contain various tags", () => {
    expect(component.find("h1").length).toEqual(1);
    expect(component.find("h2").length).toEqual(2);
    expect(component.find("a").length).toEqual(1);
  });

  test("href of a tag should point to intro", () => {
    const test_component = ReactTestUtils.renderIntoDocument(<Header />);
    const link = ReactTestUtils.findRenderedDOMComponentWithClass(
      test_component,
      "a"
    );
    expect(link.getAttribute("href").length).toEqual(3);
  });
});
