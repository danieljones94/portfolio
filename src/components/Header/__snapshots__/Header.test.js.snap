// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should match the snapshot 1`] = `
<Header>
  <header
    className="headerContainer"
  >
    <section
      className="titleContainer"
    >
      <div
        className="title"
      >
        <h1>
          DANIEL 
          <span>
            JONES
          </span>
        </h1>
        <h2>
          Web
        </h2>
        <h2>
          Developer
        </h2>
      </div>
      <a
        className="scroll"
        href="#intro"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f078",
                "M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z",
              ],
              "iconName": "chevron-down",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-down fa-w-14 "
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </a>
    </section>
  </header>
</Header>
`;
